modules/serverpod_chat/serverpod_chat_server/lib/src/endpoints/chat_endpoint.dart

```diff
@@ -287,27 +300,48 @@ class ChatEndpoint extends Endpoint {
     int? thumbHeight;
 
     try {
-      const maxImageWidth = 256;
-
       var ext = path.extension(filePath.toLowerCase());
       if ({'.jpg', '.jpeg', '.png', '.gif'}.contains(ext)) {
         var response = await http.get(url!);
         var bytes = response.bodyBytes;
-        var image = decodeImage(bytes);
-        if (image != null) {
-          if (image.width > maxImageWidth) {
-            image = copyResize(image, width: maxImageWidth);
+        // Run thumbnail generation in an isolate, because it is CPU-intensive
+        var thumbnail = await Isolate.run(() {
+          var image = decodeImage(bytes);
+          if (image == null) {
+            return null;
           }
-          var thumbPath = _generateAttachmentFilePath(userId, fileName);
+          if (image.width > maxImageWidth || image.height > maxImageHeight) {
+            // Shrink image to fit thumbnail max size
+            int? width;
+            int? height;
+            if (image.width / maxImageWidth > image.height / maxImageHeight) {
+              width = maxImageWidth;
+            } else {
+              height = maxImageHeight;
+            }
+            image = copyResize(
+              image,
+              width: width,
+              height: height,
+              interpolation: Interpolation.average,
```

Comment:
Question: Same question as previously, what was the reason for changing interpolation style? (Nearest was the default value before).

Reply:
As above.